sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -401,10 +402,17 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
       retryOptions: this._retryOptions,
       abortSignal: options?.abortSignal,
     });
+
+    if (maxMessageSize > maxBatchSizePremium) {
+      maxMessageSize = maxBatchSizePremium;
+    } else {
+      maxMessageSize = maxBatchSizeStandard;
+    }
```

Comment:
Are we speculating here whether we're connected to a premium resource? Could this be done in a more robust way?

Reply:
The maxMessageSize is from the service.  It is a good indicator since for standard it's 256 KB and for premium it is at least 1 MB.

Reply:
If the service introduces a new tier with a different batch size, would this code still work?

Reply:
We hope that existing libraries can continue to work. Does the service API use API versions to guard behavioral changes? If not, this code could be easily broken by a future change.